api/tests/routes/public/collective/endpoints/adage_mock/test_bookings.py

```diff
@@ -504,3 +510,243 @@ def test_should_raise_403_when_status_is_not_used(self, client, booking_factory,
+        expected_num_queries += 1  # 10. get (same) collective offer
+        expected_num_queries += 1  # 11. get (same) venue
+        expected_num_queries += 1  # 12. get (same) offerer
+        expected_num_queries += 1  # 13. search for offer's bookings
+        expected_num_queries += 1  # 14. get educational year (filtered)
+        expected_num_queries += 1  # 15. get redactor
+        expected_num_queries += 1  # 16. create new booking
+        expected_num_queries += 1  # 17. get (same) stock
+        expected_num_queries += 1  # 18. get (same) booking
+        expected_num_queries += 1  # 19. get (same) offer
+        expected_num_queries += 1  # 20. get (same) venue
+        expected_num_queries += 1  # 21. get (same) redactor
+        expected_num_queries += 1  # 22. get (same) institution
+        expected_num_queries += 1  # 23. get educational domain and collective domain
+        expected_num_queries += 1  # 24. get (same) offerer
+
+        with assert_num_queries(expected_num_queries):
+            self.assert_request_has_expected_result(
+                auth_client,
+                url_params={"offer_id": offer_id},
+                expected_status_code=204,
+            )
+
+        db.session.refresh(offer)
+        bookings = offer.collectiveStock.collectiveBookings
+        assert bookings
+        assert len(bookings) == 1
+
+        booking = bookings[0]
+        assert booking.status == models.CollectiveBookingStatus.PENDING
+
+    def test_cannot_book_inactive_offer(self, client):
+        plain_api_key, venue_provider = self.setup_active_venue_provider()
+        auth_client = client.with_explicit_token(plain_api_key)
+
+        deposit = factories.EducationalDepositFactory(educationalYear=factories.EducationalCurrentYearFactory())
+        offer = factories.CollectiveStockFactory(
+            collectiveOffer__provider=venue_provider.provider,
+            collectiveOffer__venue=venue_provider.venue,
+            collectiveOffer__institution=deposit.educationalInstitution,
+            collectiveOffer__isActive=False,
+        ).collectiveOffer
+
+        self.assert_request_has_expected_result(
+            auth_client,
+            url_params={"offer_id": offer.id},
+            expected_status_code=403,
+            expected_error_json={"code": "OFFER_IS_NOT_BOOKABLE"},
+        )
+
+        db.session.refresh(offer)
+        assert not offer.collectiveStock.collectiveBookings
+
+    @pytest.mark.parametrize(
+        "validation_status",
+        [OfferValidationStatus.DRAFT, OfferValidationStatus.PENDING, OfferValidationStatus.REJECTED],
+    )
+    def test_cannot_book_not_approved_offer(self, client, validation_status):
+        plain_api_key, venue_provider = self.setup_active_venue_provider()
+        auth_client = client.with_explicit_token(plain_api_key)
+
+        deposit = factories.EducationalDepositFactory(educationalYear=factories.EducationalCurrentYearFactory())
```

Comment:
pas nécéssaire car [la subFactory s'en charge](https://github.com/pass-culture/pass-culture-main/blob/master/api/src/pcapi/core/educational/factories.py#L233)
```suggestion
        deposit = factories.EducationalDepositFactory()
```

Reply:
Alors c'est ce que j'avais fait au début et... ça fonctionnera si tu ne lances que ce test ou qu'il est lancé en premier. [Autrement, chaque nouvel appel (indirect) à cette factory va créer incrémenter l'année](https://github.com/pass-culture/pass-culture-main/blob/master/api/src/pcapi/core/educational/factories.py#L198).